### PR TITLE
[IMP] Add codecov v4 question

### PR DESCRIPTION
**Pull request by rousseldenis**
Solves:

- https://github.com/OCA/oca-addons-repo-template/issues/250

### CODE REVIEW

File: copier.yml [hunk 1]
```diff
@@ -175,6 +175,12 @@ github_enable_codecov:
   help: GitHub action runs codecov/codecov-action?
   when: *ci_is_github
 
+github_v4_codecov:
```
Comment (pedrobaeza) [inline copier.yml:178]: Should this be a question? I think it should be the only option.

Comment (rousseldenis) [reply copier.yml:178]: I can, this is simpler. @sbidoul

Comment (sbidoul) [reply copier.yml:178]: Since we already have github_enable_codecov, I don't think a new question is necessary.

Comment (rousseldenis) [reply copier.yml:178]: Ok, that leads to this change :smile: :

image

Comment (sbidoul) [reply copier.yml:178]: Yeah I had not checked, so I did not know we had a question for codecov already.

File: src/.github/workflows/{% if ci == 'GitHub' %}test.yml{% endif %}.jinja [hunk 2]
```diff
@@ -145,7 +145,8 @@ jobs:
       - name: Run tests
         run: oca_run_tests
       {%- if github_enable_codecov %}
-      - uses: codecov/codecov-action@v1
+      - uses: codecov/codecov-action@v4
+      {%- endif %}
```
Comment (pedrobaeza) [inline src/.github/workflows/{% if ci == 'GitHub' %}test.yml{% endif %}.jinja:149]: This one should go out, isn't it?

**Comment by rousseldenis** [comment]
This one is ready

**Review by pedrobaeza — APPROVED** [review]
The only pending thing is if the token is needed or not, but I suppose it not too much burden to add such token globally in the organization, so better to have it at once and avoid future problems.

**Comment by celm1990** [comment]
Hi, I use Codecov for private repositories, but with `codecov-action V4`, it doesn't work with PRs from `forks` because tokenless usage is unsupported.

image

Additionally, GitHub Actions does not send secrets from forks.
https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow

image

According to this [link](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#enabling-workflows-for-forks-of-private-repositories
), I enabled these configurations in my repository's settings, but it's still not working.

image


As a workaround, for now, I am using `V3.1.6`, and it works. It's not the best practice, but do you have any other solutions?

**Pull request merged**